src/backend/distributed/executor/local_executor.c

```diff
@@ -704,42 +715,8 @@ ShouldExecuteTasksLocally(List *taskList)
 		return false;
 	}
 
-	if (CurrentLocalExecutionStatus == LOCAL_EXECUTION_REQUIRED)
+	if (GetCurrentLocalExecutionStatus() == LOCAL_EXECUTION_REQUIRED)
 	{
-		bool isValidLocalExecutionPath PG_USED_FOR_ASSERTS_ONLY = false;
-
-		/*
-		 * For various reasons, including the transaction visibility
-		 * rules (e.g., read-your-own-writes), we have to use local
-		 * execution again if it has already happened within this
-		 * transaction block.
-		 */
-		isValidLocalExecutionPath = IsMultiStatementTransaction() ||
-									InCoordinatedTransaction();
-
-		/*
-		 * In some cases, such as when a single command leads to a local
-		 * command execution followed by remote task (list) execution, we
-		 * still expect the remote execution to first try local execution
-		 * as TransactionAccessedLocalPlacement is set by the local execution.
-		 * The remote execution shouldn't create any local tasks as the local
-		 * execution should have executed all the local tasks. And, we are
-		 * ensuring it here.
-		 */
-		isValidLocalExecutionPath |= !AnyTaskAccessesLocalNode(taskList);
-
-		/*
-		 * We might error out later in the execution if it is not suitable
-		 * to execute the tasks locally.
-		 */
-		Assert(isValidLocalExecutionPath);
```

Comment:
I read the other pr too but couldn't understand why we are removing those assertions.

What additional "if" checks we need to add if we don't remove asserts ?

Reply:
an if check to see if it is a grant statement which might not be in a transaction block

as long as the transition is done with a single method, we should be fine because we have a check that prevents undesired state transtitions.

Reply:
I'm OK with this change, but lets wait until we all agree on the other PR. One suggestion could be to adjust comments instead of removing altogether.

Reply:
I  guess you should get the comment from the other PR now?